File: backend/keyServices/internal/model/Plan.go
package model

import (
	"gorm.io/gorm"
	"time"
)

type Plan struct {
	Id            int64   `json:"id" gorm:"primary_key"`     // 订阅计划ID
	GroupId       int64   `json:"group_id"`                  // 群组Id
	Name          string  `json:"name"`                      // 订阅名称
	IsSale        bool    `json:"is_sale"`                   // 启用销售
	IsRenew       bool    `json:"is_renew"`                  // 是否允许续费
	CapacityLimit int64   `json:"capacity_limit"`            // 最大用户数量限制
	Residue       int64   `json:"residue"`                   // 剩余订阅数量
	Describe      string  `json:"describe" gorm:"type:TEXT"` // 订阅计划的备注
	MonthPrice    float64 `json:"month_price"`               // 月付价格
	QuarterPrice  float64 `json:"quarter_price"`             // 季付价格
	HalfYearPrice float64 `json:"half_year_price"`           // 半年付价格
	YearPrice     float64 `json:"year_price"`                // 年付价格
	Sort          int64   `json:"sort"`                      // 排序顺序

	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `json:"deleted_at"`
}

func (Plan) TableName() string {
	return "x_plan"
}

File: nxc-frontend/src/api/admin/plan.ts
import instance from "@/axios";

interface Plan {
    id: number | null
    group_id?: number | null
    is_renew?: boolean
    is_sale?: boolean
    name: string
    sort: number | null
    capacity_limit: number | null
    residue: number | null
    describe?: string
    month_price?: number | null
    quarter_price?: number | null
    half_year_price?: number | null
    year_price?: number | null
    created_at?: string
    updated_at?: number | null
    deleted_at?: string
}

export const handleFetchPlanKv = async () => {
    try {
        let {data} = await instance.get('/api/admin/v1/plan/kv')
        return data
    } catch (err: any) {
        console.error(err)
        return false
    }
}

export const handleSubmitNewPlan = async (plan: Plan) => {
    try {
        let {data} = await instance.post('/api/admin/v1/plan', {
            ...plan,
        })
        return data;
    } catch (err: any) {
        console.error(err)
        return false
    }
}

export const handleUpdatePlan = async (plan: Plan) => {
    try {
        let {data} = await instance.put('/api/admin/v1/plan', {
            ...plan,
        })
        return data
    } catch (err: any) {
        console.error(err)
        return false
    }
}

export const handleDeletePlanById = async (planId: number) => {
    try {
        let {data} = await instance.delete('/api/admin/v1/plan', {
            params: {
                plan_id: planId,
            }
        })
        return data
    } catch (err: any) {
        console.error(err)
        return false
    }
}
//
// export const handleUpdatePlanRowIsSale = async (id: number, val: boolean) => {
//     try {
//
//     } catch (err: any) {
//         console.error(err)
//         return false
//     }
// }

export const handleUpdatePlanRowIsSaleOrRenew = async (field: 'sale' | 'renew', id: number, val: boolean) => {
    try {
        // let {data} = await instance.put(field==='sale'?'api/admin/v1/plan/sale':'/api/admin/v1/plan/renew', {
        let {data} = await instance.put(`/api/admin/v1/plan/${field === 'sale' ? 'sale' : 'renew'}`, {
            id,
            is_sale: val,
            is_renew: val,
        })
        return data
    } catch (err: any) {
        console.error(err)
        return false
    }
}